### PR TITLE
added phone number validation rules

### PR DESCRIPTION
**Pull request by alexey-reshetnik**
Added phone number validation rules based on Go implementation of libphonenumber (picked the most active repo)

### CODE REVIEW

**Comment by alexey-reshetnik** [comment]
I don't know why CLA is pending though- I signed it for all ORY repos, and so did @tepasha

**Comment by aeneasr** [comment]
Thank you, this looks great! It looks like the CLA bot is not properly detecting your signature. To fix this, try the following:

```
$ git commit  --amend --author="Author Name <email@address.com>"
```

Ensure that `Author Name` is replaced with your GitHub username (e.g. `aeneasr`) and that the email address is replaced with the email address you have set up in GitHub (e.g. `3372410+aeneasr@users.noreply.github.com`):

```
$ git commit  --amend --author="aeneasr <3372410+aeneasr@users.noreply.github.com>"
```

Once that is done, you can force-push your changes (make sure to push to the correct remote and branch!):

```
$ git push --force <remote> HEAD:<branch>
```

File: formats.go [hunk 1]
```diff
@@ -167,6 +169,32 @@ func isEmail(v interface{}) bool {
...
 }
 
+// isPhone tells whether given string is a valid phone number
+func isPhone(v interface{}) bool {
+	s, ok := v.(string)
+	if !ok {
+		return false
+	}
+
+	// entire phone number to be no more than 15 characters long
```
Comment (aeneasr) [inline formats.go:179]: Is this an international standard? If so, where could I look it up? :)

File: formats.go [hunk 2]
```diff
@@ -25,6 +26,7 @@ var Formats = map[string]func(interface{}) bool{
 	"time":                  isTime,
 	"hostname":              isHostname,
 	"email":                 isEmail,
+	"phone":                 isPhone,
```
Comment (aeneasr) [inline formats.go:29]: To avoid confusion what do you think about `phone-number` or `tel` (which matches the HTML5 tel input type)?

**Review by aeneasr — COMMENTED** [review]
Just one question and a suggestion :)

File: formats.go (hunk 1 above)
Comment (alexey-reshetnik) [reply formats.go]: Good point - removed this particular check. The library itself is performing it already.

Regarding the standard - according to the [ITU](https://en.wikipedia.org/wiki/International_Telecommunication_Union) the max len for telephone number is 15 digits. But in library that I imported max allowed len is 17 with following [explanation](https://github.com/nyaruka/phonenumbers/blob/b4d8961def2e5317064619bef19ecfb679dc80ee/phonenumbers.go#L23)

File: formats.go (hunk 2 above)
Comment (alexey-reshetnik) [reply formats.go]: Agree, renamed to tel

**Comment by alexey-reshetnik** [comment]
@aeneasr please review once again, I fixed all comments here

**Comment by aeneasr** [comment]
Thank you!

**Pull request merged**